src/main/java/com/empowerops/linqalike/DefaultedQueryable.java

```diff
@@ -588,8 +588,8 @@ boolean isDistinct(EqualityComparer<? super TElement> equalityComparer){
     }
 
     /** {@inheritDoc} */ @Override default public <TRight>
-    Queryable<Tuple<TElement, TRight>> zip(Iterable<TRight> rightElements){
-        return Linq.zip(this, rightElements, Tuple::new);
+    BiQueryable<TElement, TRight> zip(Iterable<TRight> rightElements){
+        return Linq.zip(this, rightElements);
```

Comment:
also, this _--and every other method that has an `Iterable` parameter--_ should really be expanded in the same was as `with`, `union` and `except`. **I really want a tool to do this**